Sources/Telegrammer/Bot/Telegram/Extensions/Message+Helpers.swift

```diff
@@ -82,3 +82,85 @@ public extension Message {
         return !commands.isEmpty
     }
 }
+
+// MARK: Concurrency Support
+#if compiler(>=5.5)
+@available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
+public extension Message {
+
+    /**
+     Helper method to easy reply to message
+     - Parameters:
+        - text: Text to send in reply
+        - bot: Bot from which send reply
+        - parseMode: Optional. Text format
+        - replyMarkup: Optional. Reply Markup
+     
+     - Throws: Throws on errors
+     - Returns: Future of `Message` type
+     */
+    @discardableResult
+    func reply(
+        text: String,
+        from bot: Bot,
+        parseMode: ParseMode? = nil,
+        replyMarkup: ReplyMarkup? = nil
+    ) async throws -> Message {
+        let params = Bot.SendMessageParams(
+            chatId: .chat(chat.id),
+            text: text,
+            parseMode: parseMode,
+            replyMarkup: replyMarkup
+        )
+        return try await bot.sendMessage(params: params)
```

Comment:
I'm not familiar with new swift async/await feature implementation, thus I have a question:
Is it same if we use
`return try bot.sendMess.....`
without await?
Because we don't actually need to await, we can directly return Future type.
Is this syntax correct?

Reply:
If the caller doesn't care about the result, can use:
```
Task {
   try await message.reply......
}
```
It is asynchronous and will not block.
`if we use
return try bot.sendMess.....
without await?` , we will got an error, that is 'Expression is 'async' but is not marked with 'await''.
About swift async/await: [](https://github.com/apple/swift-evolution/blob/main/proposals/0296-async-await.md)

Reply:
I think we should return the result and let the caller decide if they need to wait for it.
And if they are not called in an asynchronous function, they still use the previous Future<...>.
My english is so bad, Hope you can understand what i mean.

Reply:
Sample code: 
```Swift
func commandUpdate(_ update: Update, _ context: BotContext?) async throws {
        guard let message = update.message, let text = message.text, let entity = message.entities?.first, entity.type == .botCommand else {
            return
        }
        let msg = try await bot.sendMessage(params: Bot.SendMessageParams(chatId: .chat(message.chat.id), text: "start..."))
        app.logger.info("1")
        // need result
        let ret = try await msg.edit(text: "finish....", from: bot)
        // some code use ret.....
        app.logger.info("2")
        // don't care
        Task {
            try await msg.edit(text: "finish....3", from: bot)
        }
        app.logger.info("3")
}
```
Output:
```
[ INFO ] 1
[ INFO ] Sending request:
Request: POST https://api.telegram.org:443/botxxxxxx/editMessageText
Headers: [("Content-Type", "application/json")]
Body: Optional(AsyncHTTPClient.HTTPClient.Body(length: Optional(67), stream: (Function)))
[ INFO ] 
Response:
Code: 0
Status OK: true
Description: Empty

[ INFO ] 2
[ INFO ] 3
[ INFO ] Sending request:
Request: POST https://api.telegram.org:443/botxxxxxx/editMessageText
Headers: [("Content-Type", "application/json")]
Body: Optional(AsyncHTTPClient.HTTPClient.Body(length: Optional(68), stream: (Function)))
[ INFO ] 
Response:
Code: 0
Status OK: true
Description: Empty
```

Reply:
> I think we should return the result and let the caller decide if they need to wait for it.
> 
> And if they are not called in an asynchronous function, they still use the previous Future<...>.
> 
> My english is so bad, Hope you can understand what i mean.

I got it, thanks